### PR TITLE
Extend html_shim to expose dart:js and dart:js_util

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

File: packages/html_shim/pubspec.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ description: Shim that exposes dart:html in a Flutter application throwing excep
 author: Dart Team <misc@dartlang.org>
 homepage: https://github.com/flutter/devtools/tree/master/packages/html_shim
 
-version: 0.0.1
+version: 0.0.2
 
 environment:
   sdk: '>=2.0.0 <3.0.0'
```
Comment (terrylucas) [inline packages/html_shim/pubspec.yaml]: Should we lock the sdk to 2.5.0?

File: packages/html_shim/lib/src/_js_io.dart [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+
+/**
+ * Proxies a JavaScript Function object.
+ */
+class JsFunction extends JsObject {
+  /**
+   * Returns a [JsFunction] that captures its 'this' binding and calls [f]
+   * with the value of this passed as the first argument.
+   */
+  factory JsFunction.withThis(Function f) {
```
Comment (terrylucas) [inline packages/html_shim/lib/src/_js_io.dart]: Should this just be unsupported?

File: packages/html_shim/lib/src/_js_io.dart [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+ */
+library dart.js;
+
+import 'dart:collection' show HashMap, ListMixin;
+import 'dart:typed_data' show TypedData;
+import '_html_common_io.dart';
+
+final JsObject context = _wrapToDart(JS('', 'self'));
+
+_convertDartFunction(Function f, {bool captureThis: false}) {
```
Comment (terrylucas) [inline packages/html_shim/lib/src/_js_io.dart]: This private method isn't needed if only caller returns unsupported.

File: packages/html_shim/pubspec.yaml (hunk 1 above)
Comment (jacob314) [reply packages/html_shim/pubspec.yaml]: hmm. If there are ever any problems with this package not working with newer environments we can do that. my expectation is this package will continue to work fine with new sdk rolls assuming we aren't changing the public apis for dart:html and dart:js in breaking ways.

File: packages/html_shim/lib/src/_js_io.dart (hunk 2 above)
Comment (jacob314) [reply packages/html_shim/lib/src/_js_io.dart]: done. stripped out most of the method bodies in this file replacing them with unsupported.

File: packages/html_shim/lib/src/_js_io.dart (hunk 3 above)
Comment (jacob314) [reply packages/html_shim/lib/src/_js_io.dart]: done

**Pull request merged**